File: snippets/node/10-repository/05-delete-sord.mjs
// import the shared client:  import { connect } from "elo-playground";
// topic:    Delete an object (write, two steps)
// category: Repository & objects
// id:       repository.delete-sord

import { connect, EloError } from "elo-playground";

// --- local ELO test box (override with ELOPG_* env vars or a .env) ---
const ELO_BASE_URL = process.env.ELOPG_ELO_BASE_URL || "http://localhost:9090/ix-Repository1"; // ELOPG_DEFAULT:base_url
const ELO_USER = process.env.ELOPG_ELO_USER || "Administrator"; // ELOPG_DEFAULT:user
const ELO_PASS = process.env.ELOPG_ELO_PASSWORD || "elo"; // ELOPG_DEFAULT:password

const elo = await connect({ baseUrl: ELO_BASE_URL, user: ELO_USER, password: ELO_PASS });
const ALL = "449304431574384639";

// Provision a throwaway folder so this snippet is self-contained.
const tpl = (await elo.call("createSord", {
  parentId: "1", maskId: 0, editInfoZ: { bset: "1", sordZ: { bset: ALL } },
})).sord;
tpl.name = "pg-delete-me";
const objId = String(await elo.call("checkinSord", {
  sord: tpl, sordZ: { bset: ALL }, unlockZ: { bset: "1" },
}));
console.log("scratch folder:", objId);

try {
  // step 1 - move it to the recycle bin
  await elo.call("deleteSord", { objId, parentId: "1", deleteOptions: { deleteFinally: false } });
  console.log("step 1 (to recycle bin): ok");

  // step 2 - purge it for good
  await elo.call("deleteSord", { objId, parentId: "1", deleteOptions: { deleteFinally: true } });
  console.log("step 2 (purge):          ok");

  // confirm it is gone
  try {
    await elo.call("checkoutSord", { objId, editInfoZ: { bset: "1", sordZ: { bset: "0" } } });
    console.log("gone: false");
  } catch (e) {
    if (e instanceof EloError) console.log("gone: true");
    else throw e;
  }
} finally {
  elo.close();
}

File: snippets/python/10-repository/05-delete-sord.py
# import the shared client:  from elo_playground import connect
# topic:    Delete an object (write, two steps)
# category: Repository & objects
# id:       repository.delete-sord

from elo_playground import connect, EloError

# --- local ELO test box (override with ELOPG_* env vars or a .env) ---
ELO_BASE_URL = "http://localhost:9090/ix-Repository1"
ELO_USER = "Administrator"
ELO_PASS = "elo"

elo = connect(base_url=ELO_BASE_URL, user=ELO_USER, password=ELO_PASS)
ALL = "449304431574384639"

# Provision a throwaway folder so this snippet is self-contained.
tpl = elo.call("createSord", {"parentId": "1", "maskId": 0,
                              "editInfoZ": {"bset": "1", "sordZ": {"bset": ALL}}})["sord"]
tpl["name"] = "pg-delete-me"
obj_id = str(elo.call("checkinSord", {"sord": tpl, "sordZ": {"bset": ALL},
                                      "unlockZ": {"bset": "1"}}))
print("scratch folder:", obj_id)

try:
    # step 1 - move it to the recycle bin
    elo.call("deleteSord", {"objId": obj_id, "parentId": "1",
                            "deleteOptions": {"deleteFinally": False}})
    print("step 1 (to recycle bin): ok")

    # step 2 - purge it for good
    elo.call("deleteSord", {"objId": obj_id, "parentId": "1",
                            "deleteOptions": {"deleteFinally": True}})
    print("step 2 (purge):          ok")

    # confirm it is gone
    try:
        elo.call("checkoutSord", {"objId": obj_id,
                                  "editInfoZ": {"bset": "1", "sordZ": {"bset": "0"}}})
        print("gone: False")
    except EloError:
        print("gone: True")
finally:
    elo.close()
